include/avk/attachment.hpp

```diff
@@ -3,6 +3,39 @@
 
 namespace avk
 {
+
+	/** Describes a dynamic rendering attachment. It can only be used with pipeline that has dynamic rendering enabled.
+	 *  It has fewever parameters than regular attachment since some of its values (load/store ops etc...) are set
+	 *  when starting the dynamic render pass as opposed to being declared beforehand.
+	 *	It can describe color attachments as well as depth/stencil attachments
+	 *	and holds some additional config parameters for these attachments.
+	 */
+	struct dynamic_rendering_attachment

```

Comment:
Reusing `avk::attachment` also for dynamic pipelines would make sense insofar as in the new example application, `avk::dynamic_rendering_attachment` is used for pipeline creation... but then `avk::attachment` is required anyways in `render()`. That might be a bit confusing to framework users.

Reply:
Well, this is a very good point which _you_ bring forth! :)      
I totally agree that good API design means that it makes API misuse by the user hard.

In order to find a great solution, let me just ask you for your opinion on the following points:

**Thought 1:** How about not having a separate type `avk::dynamic_rendering_attachment`, but instead further options for `avk::attachment`? E.g. something like: 
```cpp
avk::attachment::declare_for_dynamic(...)
```

**Thought 2:** How about allowing a 
```cpp
avk::attachment::declare_for(swapchainImageView.as_reference(), avk::on_load::clear, avk::usage::color(0), avk::on_store::store);
```
_also_ for dynamic rendering pipelines, where it would serve (in contrast to non-dynamic ones) as a **default** for the respective attachment. I.e., we remember it internally and if the user does not pass a different attachment specification during rendering, we just use that default which was provided during dynamic render pipeline creation?

Reply:
I like the first one as it is a bit more explicit (and I can also use some additional checks to validate the correct usage). So if you agree I'll change it to be as you propose in Thought 1.

Reply:
> I like the first one as it is a bit more explicit (and I can also use some additional checks to validate the correct usage). So if you agree I'll change it to be as you propose in Thought 1.

Okay, let's try this! :)
I can't promise that we will not have to rework it again, but you said iterations are okay, right? 😬